src/main/java/uk/gov/hmcts/ccd/domain/types/OrgPolicyCaseAssignedRoleValidator.java

```diff
@@ -0,0 +1,79 @@
+package uk.gov.hmcts.ccd.domain.types;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import org.springframework.beans.factory.annotation.Qualifier;
+import uk.gov.hmcts.ccd.data.caseaccess.CachedCaseRoleRepository;
+import uk.gov.hmcts.ccd.data.caseaccess.CaseRoleRepository;
+import uk.gov.hmcts.ccd.data.definition.CachedCaseDefinitionRepository;
+import uk.gov.hmcts.ccd.data.definition.CaseDefinitionRepository;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseFieldDefinition;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+import javax.inject.Singleton;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.Set;
+
+@Named("OrgPolicyCaseAssignedRole")
+@Singleton
+public class OrgPolicyCaseAssignedRoleValidator implements FieldIdBasedValidator {
+
+    private final CaseDefinitionRepository caseDefinitionRepository;
+    private final CaseRoleRepository caseRoleRepository;
+    private ValidationContext validationContext;
+
+    @Inject
+    public OrgPolicyCaseAssignedRoleValidator(
+        @Qualifier(CachedCaseDefinitionRepository.QUALIFIER) final CaseDefinitionRepository caseDefinitionRepository,
+        @Qualifier(CachedCaseRoleRepository.QUALIFIER) final CaseRoleRepository caseRoleRepository
+    ) {
+        this.caseDefinitionRepository = caseDefinitionRepository;
+        this.caseRoleRepository = caseRoleRepository;
+    }
+
+    @Override
+    public String getFieldId() {
+        return CustomTypes.ORG_POLICY_CASE_ASSIGNED_ROLE.getId();
+    }
+
+    @Override
+    public List<ValidationResult> validate(String dataFieldId, JsonNode dataValue, CaseFieldDefinition caseFieldDefinition) {
+        caseFieldDefinition.getFieldTypeDefinition();
+        return validateOrganisationPolicy(dataValue, caseFieldDefinition);
+    }
+
+    @Override
+    public void setValidationContext(ValidationContext validationContext) {
+        this.validationContext = validationContext;
+    }
+
+    private List<ValidationResult> validateOrganisationPolicy(JsonNode dataValue, CaseFieldDefinition caseFieldDefinition) {
+        final String caseTypeId = validationContext.getCaseTypeId();
+        final Set<String> caseRoles = caseRoleRepository.getCaseRoles(caseTypeId);
+        final List<ValidationResult> errors = new ArrayList<>();
+        validateContent(caseFieldDefinition.getId(), dataValue, caseRoles, errors);
```

Comment:
Better to have `validateContent` create a list of errors and return it. Rather than being void